js/pro/deribit.js

```diff
@@ -0,0 +1,843 @@
+        const channel = this.safeString (params, 'channel');
+        const marketId = this.safeString (data, 'instrument_name');
+        const symbol = this.safeSymbol (marketId);
+        const timestamp = this.safeNumber (data, 'timestamp');
+        let storedOrderBook = this.safeValue (this.orderbooks, symbol);
+        if (storedOrderBook === undefined) {
+            storedOrderBook = this.countedOrderBook ();
+        }
+        const asks = this.safeValue (data, 'asks', []);
+        const bids = this.safeValue (data, 'bids', []);
+        this.handleDeltas (storedOrderBook['asks'], asks);
+        this.handleDeltas (storedOrderBook['bids'], bids);
+        storedOrderBook['nonce'] = timestamp;
+        storedOrderBook['timestamp'] = timestamp;
+        storedOrderBook['datetime'] = this.iso8601 (timestamp);
+        storedOrderBook['symbol'] = symbol;
+        this.orderbooks[symbol] = storedOrderBook;
+        client.resolve (storedOrderBook, channel);
+    }
+
+    cleanOrderBook (data) {
+        const bids = this.safeValue (data, 'bids', []);
+        const asks = this.safeValue (data, 'asks', []);
+        const cleanedBids = [];
+        for (let i = 0; i < bids.length; i++) {
+            cleanedBids.push ([ bids[i][1], bids[i][2] ]);
+        }
+        const cleanedAsks = [];
+        for (let i = 0; i < asks.length; i++) {
+            cleanedAsks.push ([ asks[i][1], asks[i][2] ]);
+        }
+        data['bids'] = cleanedBids;
+        data['asks'] = cleanedAsks;
+        return data;
+    }
+
+    handleDelta (bookside, delta) {
+        const price = delta[1];
+        const amount = delta[2];
+        if (delta[0] === 'new' || delta[0] === 'change') {
+            bookside.store (price, amount, 1);
+        } else if (delta[0] === 'delete') {
+            bookside.store (price, amount, 0);
+        }
+    }
+
+    handleDeltas (bookside, deltas) {
+        for (let i = 0; i < deltas.length; i++) {
+            this.handleDelta (bookside, deltas[i]);
+        }
+    }
+
+    async watchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name deribit#watchOrders
+         * @see https://docs.deribit.com/#user-orders-instrument_name-raw
+         * @description watches information on multiple orders made by the user
+         * @param {string} symbol unified market symbol of the market orders were made in
+         * @param {int|undefined} since the earliest time in ms to fetch orders for
+         * @param {int|undefined} limit the maximum number of  orde structures to retrieve
+         * @param {object} params extra parameters specific to the deribit api endpoint
+         * @returns {[object]} a list of [order structures]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure
+         */
+        await this.authenticate (params);
+        if (symbol !== undefined) {
+            await this.loadMarkets ();
```

Comment:
we need to load the markets regardless of having a symbol or not, it will be needed to parse orders, access currencies, etc. Can you place this above `this.authenticate` ?